htdocs/core/class/conf.class.php

```diff
@@ -144,66 +145,58 @@ function setValues($db)
 		$sql.= " ORDER BY entity";	// This is to have entity 0 first, then entity 1 that overwrite.
 
 		$resql = $db->query($sql);
-		if ($resql)
-		{
-			$i = 0;
-			$numr = $db->num_rows($resql);
-			while ($i < $numr)
+		while ($resql && $objp = $db->fetch_object($resql)) {
```

Comment:
It means we integrate in the same test a test if there is a technical error (test on $resql) and a logical condition (test on objp empty). I don't like to mix such test.
Good example here is later: the $db->free() is done whatever is the result of the technical test (test on $resql) when it should be tried only if there the $resql is not empty.
I don't think we must not work on the "theorical complexity" for such cases. This also make code not similar to other places. It also break the ability to add a else dol_print_error($db); after the test of if ($resql)